janus.ggo

```diff
@@ -2,6 +2,7 @@
 option "daemon" b "Launch Janus in background as a daemon" flag off
 option "pid-file" p "Open the specified PID file when starting Janus (default=none)" string typestr="path" optional
 option "disable-stdout" N "Disable stdout based logging" flag off
+option "log-stdout" - "Log to stdout" flag off
```

Comment:
Not sure that `-` is the right one-char shortcut here: wouldn't it be confused with what `-` usually stands for in scripts and applications? Of course, I do realize almost all letters of the alphabet are gone, we used too many :smile: 

A couple of side notes:

* For each command-line property there should be a `janus.jcfg` equivalent
* New command line properties should be documented in `janus.1` (the man file) and added to the summary in the README as well

Reply:
> Not sure that `-` is the right one-char shortcut here

`-` means "no short option". Quoting the info page of gengetopt:

```
'short'

     The short option, a single upper or lower case char, or a digit.
     If a '-' is specified, then no short option is considered for the
     long option (thus long options with no associated short options are
     allowed).  Since version 2.22 you can also specify '?' as the short
     option.
```

> For each command-line property there should be a janus.jcfg equivalent

There is already a `log_to_stdout` options in `janus.jcfg` but it defaults to `true` while the command line flag defaults to `false`. This issue exists only because I had to preserve the current behaviour where daemonize would disable stdout even if `log_to_stdout` was set to `true`.

Should I just rename the command line flag to `--log-to-stdout` instead of `--log-stdout` and leave everything else as-is?

Reply:
> `-` means "no short option". Quoting the info page of gengetopt:

Ah thanks for the heads up, I didn't know about that!

> There is already a `log_to_stdout` options in `janus.jcfg` but it defaults to `true` while the command line flag defaults to `false`. This issue exists only because I had to preserve the current behaviour where daemonize would disable stdout even if `log_to_stdout` was set to `true`.

In that case I think it may be worth making it clearer in the description of the property should that this is a way to make sure stdout to remain alive even, or especially, when demonizing (since that's basically the only thing it's used for). The "Log to stdout" text doesn't make that clear enough IMO.

> Should I just rename the command line flag to `--log-to-stdout` instead of `--log-stdout` and leave everything else as-is?

Makes sense!

Reply:
I did change the `--log-stdout` flag to `--log-to-stdout` in the end. That's because the analogous configuration `log_to_file` has the matching flag `--log-file` and I think it makes sense to keep it consistent.

Reply:
Are you sure the right changes were pushed? I still see the old name. Besides I see the updated description in the man-file, but notin the .ggo one?

Reply:
That's what I was explaining.

I thought that changing `--log-stdout` to `--log-to-stdout` was not consistent with `--log-file` that is NOT named `--log-to-file` despite its configuration option being `log_to_file`.

So I did NOT change anything here. Do you still want to rename it even in light of this argument?

Reply:
Sorry, maybe you were referring to the description and not the name. I just pushed the update for that as well.

Reply:
> Do you still want to rename it even in light of this argument?

No, I'm fine with that. I asked because you wrote:

> I did change the --log-stdout flag to --log-to-stdout in the end.

but the name was still the old one. Did you mean "I didn't change... in the end"?
Description looks good, thanks!

Reply:
> Did you mean "I didn't change... in the end"?

Oh, yeah, didn't notice the typo. Sorry about that.